### DISCUSSION TITLE
"docker run" & "docker logs" not outputting anything

### DISCUSSION THREAD

**Discussion by KaKi87** [caprover/caprover#1065] [Q&A] [answered]
Hello,

Here are my shell commands :

```bash
$ docker run -p 3011:80 -p 3012:443 -p 3010:3000 -v /var/run/docker.sock:/var/run/docker.sock -v /home/kaki/opt/cr.kaki87.net:/captain caprover/caprover -e BY_PASS_PROXY_CHECK=true
# -> empty
$ docker ps -a
CONTAINER ID   IMAGE               COMMAND                  CREATED          STATUS                     PORTS     NAMES
a40f00234c0f   caprover/caprover   "docker-entrypoint.s…"   12 seconds ago   Exited (0) 7 seconds ago             focused_benz
$ docker logs a40f00234c0f
# -> empty
```

Here are my apache virtual hosts :

```
<VirtualHost *:80>
    ServerName cr.kaki87.net
    RewriteEngine On
    RewriteCond %{HTTPS} off
    RewriteRule (.*) https://%{HTTP_HOST}%{REQUEST_URI}
</VirtualHost>

<VirtualHost *:443>
    ServerName cr.kaki87.net
    ProxyRequests Off
    ProxyPass / http://localhost:3010/
    ProxyPassReverse / http://localhost:3010/
    SSLCertificateFile /home/kaki/.acme.sh/kaki87.net/fullchain.cer
    SSLCertificateKeyFile /home/kaki/.acme.sh/kaki87.net/kaki87.net.key
</VirtualHost>

<VirtualHost *:80>
    ServerName _.cr.kaki87.net
    ServerAlias *.cr.kaki87.net
    ProxyRequests Off
    ProxyPass / http://localhost:3011/
    ProxyPassReverse / http://localhost:3011/
</VirtualHost>

<VirtualHost *:443>
    ServerName _.cr.kaki87.net
    ServerAlias *.cr.kaki87.net
    ProxyRequests Off
    ProxyPass / http://localhost:3012/
    ProxyPassReverse / http://localhost:3012/
    SSLCertificateFile /home/kaki/.acme.sh/kaki87.net/fullchain.cer
    SSLCertificateKeyFile /home/kaki/.acme.sh/kaki87.net/kaki87.net.key
</VirtualHost>
```

Thanks

**Comment by githubsaturn** [comment, marked as answer]
HI @KaKi87  - CapRover does not work on other ports. When you start the installation container, what happens under the hood is that:
- Docker swarm gets initialized
- A few docker services get created
- The installation container die

**Reply by KaKi87** [reply]
So I can't make CapRover coexist with an apache server ?
You mean it's an installation container, so there's another one which acts as a runtime container ?

**Reply by githubsaturn** [reply]
> So I can't make CapRover coexist with an apache server ?

Yes

> You mean it's an installation container, so there's another one which acts as a runtime container ?

Yes.

**Reply by KaKi87** [reply]
Why does CapRover cares about the host port, doesn't Docker provide abstraction for it ?

**Reply by githubsaturn** [reply]
In theory it is possible. But we never got around it to change the code as it is not a common use case for our users. The main use case of CapRover is installation on a brand new server. Feel free to open a PR and change that. There are multiple surfaces that need to be updated:
- Healthcheck
- Certbot (CapRover creates and manages the SSL)
- Nginx

**Reply by KaKi87** [reply]
I understand, here is my use case : I have a dedicated server, whose resources I do not fully use, therefore I would like to allow other people to host their own stuff on it, without giving them direct access to the machine.

If said updates were performed, would CapRover be viable for my use case ?

If yes, which experience would I need to perform those updates ?

**Reply by githubsaturn** [reply]
> If said updates were performed, would CapRover be viable for my use case ?

No. One issue is the port collision that I mentioned above. The other issue is that, giving CapRover access to people is essentially equivalent of giving them full access to the machine. You can map volumes `/` to `/host` in the container and see the entire content of the machine.

**Reply by KaKi87** [reply]
I see, thank you for your answer.
